book/src/dev/rfcs/0011-value-pool.md

```diff
@@ -0,0 +1,229 @@
+        i64::from_be_bytes(array).try_into().unwrap()
+    }
+}
+```
+
+## Changes to `zebra-state/src/service/finalized_state.rs`
+
+First we add columns for each pool:
+
+```rust
+rocksdb::ColumnFamilyDescriptor::new("transparent_pool", db_options.clone()),
+rocksdb::ColumnFamilyDescriptor::new("sprout_pool", db_options.clone()),
+rocksdb::ColumnFamilyDescriptor::new("sapling_pool", db_options.clone()),
+rocksdb::ColumnFamilyDescriptor::new("orchard_pool", db_options.clone()),
+```
+
+At block commit(`commit_finalized_direct()`) we crete the handles for the new columns:
+
+```rust
+let transparent_pool = self.db.cf_handle("transparent_pool").unwrap();
+let sprout_pool = self.db.cf_handle("sprout_pool").unwrap();
+let sapling_pool = self.db.cf_handle("sapling_pool").unwrap();
+let orchard_pool = self.db.cf_handle("orchard_pool").unwrap();
+```
+
+Next we save each value pool for each upcoming block:
+
+```rust
+let mut current_transparent_pool = Amount::try_from(0).unwrap();
+let mut current_sprout_pool = Amount::try_from(0).unwrap();
+let mut current_sapling_pool = Amount::try_from(0).unwrap();
+let mut current_orchard_pool = Amount::try_from(0).unwrap();
+
+if height > block::Height(1) {
+    current_transparent_pool = self.pool(Pool::Transparent).unwrap().1;
+    current_sprout_pool = self.pool(Pool::Sprout).unwrap().1;
+    current_sapling_pool = self.pool(Pool::Sapling).unwrap().1;
+    current_orchard_pool = self.pool(Pool::Orchard).unwrap().1;
+}
+
+batch.zs_insert(transparent_pool, height, (current_transparent_pool + block.value_pool(Pool::Transparent)).unwrap());
+batch.zs_insert(sprout_pool, height, (current_sprout_pool + block.value_pool(Pool::Sprout)).unwrap());
+batch.zs_insert(sapling_pool, height, (current_sapling_pool + block.value_pool(Pool::Sapling)).unwrap());
+batch.zs_insert(orchard_pool, height, (current_orchard_pool + block.value_pool(Pool::Orchard)).unwrap());
+```
+
+Note that each pool is a sum of the amount in the last block plus the amount in the upcoming current block. Each block must have a value pool amount in the previous block (can be 0) except for the genesis block.
+
+The `pool()` function will get the value of the pool at the tip as follows:
+
+```rust
+/// Returns the requested pool at tip.
+pub fn pool(&self, pool_name: Pool) -> Option<(block::Height, Amount)> {
+    let pool = match pool_name {
+        Pool::Transparent => self.db.cf_handle("transparent_pool").unwrap(),
+        Pool::Sprout => self.db.cf_handle("sprout_pool").unwrap(),
+        Pool::Sapling => self.db.cf_handle("sapling_pool").unwrap(),
+        Pool::Orchard => self.db.cf_handle("orchard_pool").unwrap(),
+    };
+    self.db
+        .iterator_cf(pool, rocksdb::IteratorMode::End)
+        .next()
+        .map(|(height_bytes, amount_bytes)| {
+            let height = block::Height::from_bytes(height_bytes);
+            let amount = Amount::from_bytes(amount_bytes);
+
+            (height, amount)
+        })
+}
+```
+
```

Comment:
Please add a Tree/Key/Value table for any new column families:
https://github.com/ZcashFoundation/zebra/blob/main/book/src/dev/rfcs/0005-state-updates.md#rocksdb-data-structures